### PR TITLE
support multiple nodes of a graph, apply different endpoint env for different node

### PR DESCRIPTION
**Pull request by KfreeZ**
## Description

#166 

this PR also adopt the new manifest, when new manifest is merged, some unnecessary code will be removed for simplicity

mark it as draft

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: microservices-connector/config/samples/chatQnA_xeon.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ spec:
           serviceName: embedding-svc
           config:
             endpoint: /v1/embeddings
+            TEI_EMBEDDING_ENDPOINT: TeiEmbedding
```
Comment (irisdingbj) [inline microservices-connector/config/samples/chatQnA_xeon.yaml:26]: This should use servcieName, in this case `tei-embedding-svc`.

File: microservices-connector/config/samples/chatQnA_xeon.yaml [hunk 2]
```diff
@@ -33,6 +34,7 @@ spec:
           serviceName: retriever-svc
           config:
             endpoint: /v1/retrieval
+            REDIS_URL: VectorDB
```
Comment (irisdingbj) [inline microservices-connector/config/samples/chatQnA_xeon.yaml:37]: same here, use `redis-vector-db`, in this way we can support multiple endpoint for switch scenario

File: microservices-connector/config/samples/chatQnA_xeon.yaml [hunk 3]
```diff
@@ -43,6 +45,7 @@ spec:
           serviceName: reranking-svc
           config:
             endpoint: /v1/reranking
+            TEI_RERANKING_ENDPOINT: TeiReranking
```
Comment (irisdingbj) [inline microservices-connector/config/samples/chatQnA_xeon.yaml:48]: the same

File: microservices-connector/config/samples/chatQnA_xeon.yaml [hunk 4]
```diff
@@ -55,6 +58,7 @@ spec:
           serviceName: llm-svc
           config:
             endpoint: /v1/chat/completions
+            TGI_LLM_ENDPOINT: Tgi
```
Comment (irisdingbj) [inline microservices-connector/config/samples/chatQnA_xeon.yaml:61]: the same

File: microservices-connector/config/samples/chatQnA_xeon.yaml (hunk 2 above)
Comment (KfreeZ) [reply microservices-connector/config/samples/chatQnA_xeon.yaml:37]: In the multiple endpoint case, is it better to use different step name, like llm-big, llm-small?

**Review by mkbhanda — APPROVED** [review]
LGTM

**Comment by KfreeZ** [comment]
use #203 to commit code

**Pull request closed without merging**